openpype/modules/deadline/plugins/publish/submit_maya_remote_publish_deadline.py

```diff
@@ -0,0 +1,134 @@
+
+    def process(self, instance):
+        settings = get_project_settings(os.getenv("AVALON_PROJECT"))
+        # use setting for publish job on farm, no reason to have it separately
+        deadline_publish_job_sett = (settings["deadline"]
+                                     ["publish"]
+                                     ["ProcessSubmittedJobOnFarm"])
+
+        # Ensure no errors so far
+        assert (all(result["success"]
+                    for result in instance.context.data["results"]),
+                    ("Errors found, aborting integration.."))
+
+        if not instance.data["publish"]:
+            self.log.warning("No active instances found. "
+                             "Skipping submission..")
+            return
+
+        scene = instance.context.data["currentFile"]
+        scenename = os.path.basename(scene)
+
+        # Get project code
+        project_name = legacy_io.Session["AVALON_PROJECT"]
+
+        job_name = "{scene} [PUBLISH]".format(scene=scenename)
+        batch_name = "{code} - {scene}".format(code=project_name,
+                                               scene=scenename)
+
+        # Generate the payload for Deadline submission
+        payload = {
+            "JobInfo": {
+                "Plugin": "MayaBatch",
+                "BatchName": batch_name,
+                "Name": job_name,
+                "UserName": instance.context.data["user"],
+                "Comment": instance.context.data.get("comment", ""),
+                # "InitialStatus": state
+                "Department": deadline_publish_job_sett["deadline_department"],
+                "ChunkSize": deadline_publish_job_sett["deadline_chunk_size"],
+                "Priority": deadline_publish_job_sett["deadline_priority"],
+                "Group": deadline_publish_job_sett["deadline_group"],
+                "Pool": deadline_publish_job_sett["deadline_pool"],
+            },
+            "PluginInfo": {
+
+                "Build": None,  # Don't force build
+                "StrictErrorChecking": True,
+                "ScriptJob": True,
+
+                # Inputs
+                "SceneFile": scene,
+                "ScriptFilename": "{OPENPYPE_REPOS_ROOT}/openpype/scripts/remote_publish.py",   # noqa
+
+                # Mandatory for Deadline
+                "Version": cmds.about(version=True),
+
+                # Resolve relative references
+                "ProjectPath": cmds.workspace(query=True,
+                                              rootDirectory=True),
+
+            },
+
+            # Mandatory for Deadline, may be empty
+            "AuxFiles": []
+        }
+
+        # Include critical environment variables with submission + api.Session
+        keys = [
+            "FTRACK_API_USER",
+            "FTRACK_API_KEY",
+            "FTRACK_SERVER"
+        ]
+        environment = dict({key: os.environ[key] for key in keys
+                            if key in os.environ}, **legacy_io.Session)
```

Comment:
Doesn't this already insert the full `legacy_io.Session` into `environment`?

Reply:
I don't think anyone knows why is this in all DL submit jobs, so it might be cleaned up a bit, but I would keep it to another PR, as it would mean to retest all publishes to DL for all the implemented hosts.